files.go

```diff
@@ -506,20 +506,28 @@ func (api *Client) ShareFilePublicURLContext(ctx context.Context, fileID string)
 	return &response.File, response.Comments, &response.Paging, nil
 }
 
-// getUploadURLExternal gets a URL and fileID from slack which can later be used to upload a file.
-func (api *Client) getUploadURLExternal(ctx context.Context, params getUploadURLExternalParameters) (*getUploadURLExternalResponse, error) {
+// GetUploadURLExternalContext gets a URL and fileID from slack which can later be used to upload a file.
+// Slack API docs: https://api.slack.com/methods/files.getUploadURLExternal
+func (api *Client) GetUploadURLExternalContext(ctx context.Context, params GetUploadURLExternalParameters) (*GetUploadURLExternalResponse, error) {
+	if params.FileName == "" {
+		return nil, fmt.Errorf("GetUploadURLExternalContext: fileName cannot be empty")
```

Comment:
Do you mind removing `GetUploadURLExternalContext` from the error? Happy for the message to indicate where it's at but this format doesn't quite fit with everything else.

I am aware we already have `file.upload.v2:` in `fmt.Errorf` in this file so happy if you want to change to be sort of that format but what I'd really prefer is to not have the name of the function in the error returned.

Reply:
Thank you. I’ve updated it to include the API name instead of the function name.

https://github.com/slack-go/slack/pull/1376/commits/9a7fd29f8ccb6698f14e20a2f418d3e27012b439